pkg/translate/field_mask_test.go

```diff
@@ -0,0 +1,282 @@
+package translate
+
+import (
+	"testing"
+
+	"github.com/brianvoe/gofakeit/v6"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.expect.digital/translate/pkg/model"
+	"golang.org/x/exp/maps"
+)
+
+func mergeSlices[T ~[]E, E any](a, b T) T {
```

Comment:
Why not simply `append(a, b...)`? If there is enough capacity, no malloc is done.

Reply:
Potential side effects, I wanted to be 100% sure that merging slices and maps is pure, e.g. returns merged without modifying a or b
> I would like to emphasize @icza answer and simplify it a bit since it is a crucial concept. I assume that reader is familiar with slices.
c := append(a, b...)
This is a valid answer to the question. BUT if you need to use slices 'a' and 'c' later in code in different context, this is not the safe way to concatenate slices.

https://stackoverflow.com/a/58726780/19159194